src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java

```diff
@@ -108,45 +109,56 @@ private void processQuery(JobContext context) throws IOException, InterruptedExc
     }
   }
 
-  private String generateQuery(String partitionFromDate, String partitionToDate, String project, String dataset,
-                               String table, @Nullable String serviceFilePath) {
-    if (partitionFromDate == null && partitionToDate == null) {
+  private String generateQuery(String partitionFromDate, String partitionToDate, String filter,
+                               String project, String dataset, String table, @Nullable String serviceFilePath) {
+    if (partitionFromDate == null && partitionToDate == null && filter == null) {
       return null;
     }
     String queryTemplate = "select * from %s where %s";
     com.google.cloud.bigquery.Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, table,
                                                                                 serviceFilePath);
     StandardTableDefinition tableDefinition = Objects.requireNonNull(sourceTable).getDefinition();
     TimePartitioning timePartitioning = tableDefinition.getTimePartitioning();
-    if (timePartitioning == null) {
+    if (timePartitioning == null && filter == null) {
       return null;
     }
     StringBuilder condition = new StringBuilder();
-    String columnName = timePartitioning.getField() != null ? timePartitioning.getField() : DEFAULT_COLUMN_NAME;
 
-    LegacySQLTypeName columnType = null;
-    if (!DEFAULT_COLUMN_NAME.equals(columnName)) {
-      columnType = tableDefinition.getSchema().getFields().get(columnName).getType();
-    }
+    if (timePartitioning != null) {
```

Comment:
try to avoid nesting large chunks of code in an if statement, it makes it difficult to read and understand. In this particular case, it would be good to call another method that just generates the time part of the where clause.